core/opendaq/opendaq/include/opendaq/instance_builder.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/stringobject.h>
+#include <opendaq/instance.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_devices
+ * @addtogroup opendaq_instance InstanceBuilder
+ * @{
+ */
+
+/*!
+ * @brief Builder component of Instance objects. Contains setter methods to configure the Instance parameters, and a
+ * `build` method that builds the Instance object.
+ */
+DECLARE_OPENDAQ_INTERFACE(IInstanceBuilder, IBaseObject)
+{
+    /*!
+     * @brief Builds and returns a Instance object using the currently set values of the Builder.
+     * @param[out] instance The built Instance.
+     */
+    virtual ErrCode INTERFACE_FUNC build(IInstance** instance) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the Logger of Instance
+     * @param logger The Logger of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC setLogger(ILogger* logger) = 0;
+
+    /*!
+     * @brief Gets the Logger of Instance
+     * @param[out] logger Tthe Logger of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC getLogger(ILogger** logger) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the Logger global level of Instance
+     * @param logLevel The Logger global level of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC setGlobalLogLevel(LogLevel logLevel) = 0;
+
+    /*!
+     * @brief Gets the Logger global level of Instance
+     * @param[out] logLevel The Logger global level of Instance
+     */
+    virtual ErrCode INTERFACE_FUNC getGlobalLogLevel(LogLevel* logLevel) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the Logger level of Instance component
+     * @param component The name of Instance component
+     * @param logLevel The log level of Instance component
+     */
+    virtual ErrCode INTERFACE_FUNC setComponentLogLevel(IString* component, LogLevel logLevel) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Sets the sink logger level of default Instance logger. Ignored if was set custom logger
```

Comment:
Grammar in comment -> "Ignored if a custom logger has already been set". Similar in other comments.